### PR TITLE
SelfHostedEtcd: get etcd service IP from APIServer

### PR DESCRIPTION
**Pull request by hongchaodeng**
Instead of parsing from static file, we change it to get etcd service IP from APIServer.

fix https://github.com/kubernetes-incubator/bootkube/issues/508

### CODE REVIEW

**Comment by hongchaodeng** [comment]
@aaronlevy @xiang90

File: pkg/asset/k8s.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ import (
 )
 
 const (
+	EtcdServiceName = "etcd-service"
```
Comment (xiang90) [inline pkg/asset/k8s.go]: comment on public const?

File: pkg/util/etcdutil/migrate.go [hunk 2]
```diff
@@ -62,6 +64,11 @@ func Migrate(kubeConfig clientcmd.ClientConfig, etcdServiceIP string) error {
 	}
 	glog.Info("etcd cluster for migration is now running")
 
+	etcdServiceIP, err := getServiceIP(kubecli, api.NamespaceSystem, asset.EtcdServiceName)
```
Comment (xiang90) [inline pkg/util/etcdutil/migrate.go]: can we do this at line51? it is cleaner to get all prerequisites at one place before we start the migration.

**Comment by xiang90** [comment]
looks good to me.

defer to other cc'd people. @dghubble is the author of this piece of code.

File: pkg/util/etcdutil/migrate.go (hunk 2 above)
Comment (hongchaodeng) [inline pkg/util/etcdutil/migrate.go]: Isn't the code clearer to the variable closer to where it is used?

**Comment by hongchaodeng** [comment]
fixed. PTAL

File: pkg/util/etcdutil/migrate.go (hunk 2 above)
Comment (xiang90) [inline pkg/util/etcdutil/migrate.go]: if this is a simple variable declaration, then yes. but this is a func call that does some initializations.

**Comment by hongchaodeng** [comment]
@aaronlevy @xiang90

**Comment by dghubble** [comment]
Seems fine to me. Detecting the etcd IP from the service was just a means of getting away from the old hardcoded IP.

**Comment by hongchaodeng** [comment]
fixed

**Comment by xiang90** [comment]
LGTM

**Comment by hongchaodeng** [comment]
rktbot run tests

ssh failure:
```
 run.go:147: creating cluster: populating cache: copying public keys: ssh: handshake failed: EOF: 
```

**Comment by hongchaodeng** [comment]
rktbot run tests

**Comment by hongchaodeng** [comment]
rktbot run tests

**Comment by hongchaodeng** [comment]
@aaronlevy 
Can you approve this?

**Pull request merged**